spec/granite/form/model/validations/nested_spec.rb

```diff
@@ -114,35 +114,75 @@
 
   context 'object field is invalid and referenced object does not include AutosaveAssociation' do
     before do
-      stub_class(:validated_object) do
-        include Granite::Form::Model::Validations
-        include Granite::Form::Model::Attributes
-        include Granite::Form::Model::Primary
-
-        primary :id
+      stub_model(:validated_object) do
         attribute :title, String
         validates_presence_of :title
       end
 
       Main.class_eval do
-        include Granite::Form::Model::Associations
         attribute :object, Object
         validates :object, nested: true
       end
     end
 
-    subject(:instance) { Main.instantiate name: 'hello', object: object, validated_one: {name: 'name'} }
+    subject(:instance) { Main.instantiate name: 'hello', object: object }
+
+    let(:validation_options) { {} }
 
-    context do
+    context 'nested object is valid' do
       let(:object) { ValidatedObject.new(title: 'Mr.') }
+
       it { is_expected.to be_valid }
     end
 
-    context do
+    context 'nested object is invalid' do
       let(:object) { ValidatedObject.new }
+
       it do
         expect { subject.valid? }.not_to raise_error
         expect(subject).not_to be_valid
+        expect(subject.errors.count).to eq(1)
+      end
+
+      context 'nested validation runs twice' do
+        before do
+          Main.class_eval do
+            validates :object, nested: true # Adds existing validation second time
+          end
+        end
+
+        it do
+          subject.validate
+          expect(subject.errors.count).to eq(1)
+        end
+
+        context 'nested object validation has condition' do
```

Comment:
I still don't understand why this spec is needed? Can you tell me how would you rewrite `import_errors` to make only this spec fail?

Reply:
Something like:
```ruby
def self.import_errors(from, to, prefix)
  from.each do |error|
    key = "#{prefix}.#{error.attribute}"
    next if to.added?(key, error.type, error.options.except(*::ActiveModel::Error::MESSAGE_OPTIONS))
    to.import(error, attribute: key)
  end
end
```
I mean to me it looks like callback options, e.g. conditions, and message are from different types of options so I wanted to highlight that we cover both cases. If you still think it's redundant I'll remove this spec.

Reply:
OK let's keep it.